### PR TITLE
Improve log function

### LINKED ISSUE #1169: log-system-out writes "raw" query to std out too
when `util:log-system-out("hahaha")`  is called from a not-stored-query, also the query itself is printed to standard out, giving a lot of noise.

e.g. the above yields into 

```
(Line: 1 util:log-system-out("hahaha")) hahaha
```

whilst I'd expect

```
(Line: 1 String) hahaha
```

the cause is in org.exist.xquery.AbstractExpression#getSource line 121:

```java
if(getSource() != null && getSource().getKey() != null) {
  buf.append(" ");
  buf.append(getSource().getKey());
}
```

as far as I see `getSource().getKey()` returns the whole query string, which does not make sense to me?

### PR DESCRIPTION
**Pull request by dizzzz**
Fixes #1169

### CODE REVIEW

File: src/org/exist/xquery/functions/util/LogFunction.java [hunk 1]
```diff
@@ -173,10 +174,7 @@ public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathExce
                 final String logname = args[1].getStringValue();
 
                 // Use specific logger when provided
-                Logger logger = LOG;
-                if (logname != null && logname.length() > 0) {
-                    logger = LogManager.getLogger(logname);
-                }
+                Logger logger = StringUtils.isBlank(logname) ? LOG : LogManager.getLogger(logname);
```
Comment (adamretter) [inline src/org/exist/xquery/functions/util/LogFunction.java]: Do we really need `StringUtils.isBlank(logname)`?  I would rather `logname == null || logname.isEmpty()`

File: src/org/exist/xquery/functions/util/LogFunction.java [hunk 2]
```diff
@@ -3,191 +3,213 @@
...
+    };
+
+    public LogFunction(XQueryContext context, FunctionSignature signature) {
+        super(context, signature);
+    }
+
+    /* (non-Javadoc)
+     * @see org.exist.xquery.BasicFunction#eval(org.exist.xquery.value.Sequence[], org.exist.xquery.value.Sequence)
+     */
+    public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathException {
```
Comment (adamretter) [inline src/org/exist/xquery/functions/util/LogFunction.java]: The junk Eclipse javadoc comment should be replaced with `@Override` whilst you are cleaning up.

File: src/org/exist/xquery/functions/util/LogFunction.java [hunk 3]
```diff
@@ -3,191 +3,213 @@
...
-}
+    /**
+     *  Write log message to logger with a priority.
+     *
+     * @param buffer    The log text
+     * @param loglevel  The priority of the log message
+     * @param logger    The actual logger
+     */
+    private void writeLog(StringBuilder buffer, String loglevel, Logger logger) {
+        switch (loglevel) {
```
Comment (adamretter) [inline src/org/exist/xquery/functions/util/LogFunction.java]: These variables could be `final`

File: src/org/exist/xquery/functions/util/LogFunction.java (hunk 1 above)
Comment (dizzzz) [inline src/org/exist/xquery/functions/util/LogFunction.java]: it is essentially the same right? I use this class basically always..... for readability.
The isBlank() also checks for whitespace characters....

Comment (dizzzz) [inline src/org/exist/xquery/functions/util/LogFunction.java]: ok, done

File: src/org/exist/xquery/functions/util/LogFunction.java (hunk 2 above)
Comment (dizzzz) [inline src/org/exist/xquery/functions/util/LogFunction.java]: ok, done

File: src/org/exist/xquery/functions/util/LogFunction.java (hunk 3 above)
Comment (dizzzz) [inline src/org/exist/xquery/functions/util/LogFunction.java]: done

**Pull request merged**